AdyenComponents/BACS Direct Debit/Scenes/Input/BACSDirectDebitInputPresenter.swift

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+
+internal protocol BACSDirectDebitPresenterProtocol: AnyObject {
+    func viewDidLoad()
+}
+
+internal class BACSDirectDebitPresenter: BACSDirectDebitPresenterProtocol {
+
+    // MARK: - Properties
+
+    private let view: BACSDirectDebitInputFormViewProtocol
+    private let router: BACSDirectDebitRouterProtocol
+    private let itemsFactory: BACSDirectDebitItemsFactory
+
+    // MARK: - Items
+
+    private var holderNameItem: FormTextInputItem?
+    private var bankAccountNumberItem: FormTextInputItem?
+    private var sortCodeItem: FormTextInputItem?
+    private var emailItem: FormTextInputItem?
+    private var amountConsentToggleItem: FormToggleItem?
+    private var legalConsentToggleItem: FormToggleItem?
+    private var continueButtonItem: FormButtonItem?
+
+    // MARK: - Initializers
+
+    internal init(view: BACSDirectDebitInputFormViewProtocol,
+                  router: BACSDirectDebitRouterProtocol,
+                  itemsFactory: BACSDirectDebitItemsFactory) {
+        self.view = view
+        self.router = router
+        self.itemsFactory = itemsFactory
+        setupItems()
+        setupView()
+    }
+
+    // MARK: - BACSDirectDebitPresenterProtocol
+
+    internal func viewDidLoad() {
+        // TODO: - Complete logic
+    }
+
+    // MARK: - Private
+
+    private func setupItems() {
+        holderNameItem = itemsFactory.createHolderNameItem()
+        bankAccountNumberItem = itemsFactory.createBankAccountNumberItem()
+        sortCodeItem = itemsFactory.createSortCodeItem()
+        emailItem = itemsFactory.createEmailItem()
+        amountConsentToggleItem = itemsFactory.createAmountConsentToggle()
+        legalConsentToggleItem = itemsFactory.createLegalConsentToggle()
+
+        continueButtonItem = itemsFactory.createContinueButton()
+        continueButtonItem?.buttonSelectionHandler = continuePayment
+    }
+
+    private func setupView() {
+        view.add(item: holderNameItem)
+        view.add(item: bankAccountNumberItem)
+        view.add(item: sortCodeItem)
+        view.add(item: emailItem)
+        view.add(item: FormSpacerItem(numberOfSpaces: 2))
+        view.add(item: amountConsentToggleItem)
+        view.add(item: FormSpacerItem(numberOfSpaces: 1))
+        view.add(item: legalConsentToggleItem)
+        view.add(item: FormSpacerItem(numberOfSpaces: 2))
+        view.add(item: continueButtonItem)
+        view.add(item: FormSpacerItem(numberOfSpaces: 1))
+    }
+
+    private var isFormValid: Bool {
```

Comment:
Maybe call `validate()` in this function instead of `displayValidation` and remove that one?